src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -554,29 +554,27 @@ private BoundPattern BindRecursivePattern(RecursivePatternSyntax node, TypeSymbo
                 }
                 for (int i = 0; i < node.Subpatterns.Count; i++)
                 {
-                    var subPattern = node.Subpatterns[i];
+                    var subpatternSyntax = node.Subpatterns[i];
                     bool isError = i >= elementTypes.Length;
                     TypeSymbol elementType = isError ? CreateErrorType() : elementTypes[i];
                     FieldSymbol foundField = null;
-                    if (subPattern.NameColon != null)
+                    if (subpatternSyntax.NameColon != null)
                     {
-                        string name = subPattern.NameColon.Name.Identifier.ValueText;
-                        foundField = CheckIsTupleElement(subPattern.NameColon.Name, (NamedTypeSymbol)declType, name, i, diagnostics);
+                        string name = subpatternSyntax.NameColon.Name.Identifier.ValueText;
+                        foundField = CheckIsTupleElement(subpatternSyntax.NameColon.Name, (NamedTypeSymbol)declType, name, i, diagnostics);
                     }
 
                     BoundSubpattern boundSubpattern = new BoundSubpattern(
-                        subPattern,
+                        subpatternSyntax,
                         foundField,
-                        BindPattern(subPattern.Pattern, elementType, isError, diagnostics));
+                        BindPattern(subpatternSyntax.Pattern, elementType, isError, diagnostics));
                     patterns.Add(boundSubpattern);
                 }
             }
             else
             {
                 // It is not a tuple type. Seek an appropriate Deconstruct method.
                 var inputPlaceholder = new BoundImplicitReceiver(node, declType); // A fake receiver expression to permit us to reuse binding logic
-                // PROTOTYPE(patterns2): Can we include element names node.Subpatterns[i].NameColon?.Name in the AnalyzedArguments
```

Comment:
>PROTOTYPE [](start = 19, length = 9)

I assume we have a test for this scenario (two Deconstruct methods, deconstruction pattern is ambiguous despite names).
Did LDM make a decision on this (in the scope of C# 8.0)?

Reply:
This is an open issue in [https://github.com/dotnet/csharplang/issues/1054](https://github.com/dotnet/csharplang/issues/1054)

---
In reply to: [191023469](https://github.com/dotnet/roslyn/pull/27160#discussion_r191023469) [](ancestors = 191023469)